nixos/modules/services/scheduling/scx.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.scx;
+in
+{
+  options.services.scx = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable SCX service, a scheduler daemon with a wide variety of
+        scheduling algorithms to improve system performance.
+
+        ::: {.note}
+        This service requires a kernel with the Sched-ext feature. Generally, kernel version 6.12 and later are supported. See
+        [kernel documentation](https://www.kernel.org/doc/html/v6.12-rc1/scheduler/sched-ext.html#switching-to-and-from-sched-ext)
```

Comment:
How does this documentation actually help?

Reply:
I added in the hopes of a clueless user finding these:

Oh I don't have the required kernel, let's see oh I need to build the kernel with these options.

Reply:
A user clueless enough to see an assertion about their kernel config and not understand what it means should not be allowed near custom kernels for their own safety.